File: tests/fixtures/simple.ex
# Elixir test fixture
defmodule Point do
  @moduledoc "A point in 2D space"
  
  @version "1.0.0"
  
  defstruct x: 0.0, y: 0.0
  
  def new(x, y) do
    %Point{x: x, y: y}
  end
  
  def distance(%Point{x: x1, y: y1}, %Point{x: x2, y: y2}) do
    dx = x1 - x2
    dy = y1 - y2
    :math.sqrt(dx * dx + dy * dy)
  end
  
  defp private_helper(value) do
    value * 2
  end
end

defprotocol Drawable do
  def draw(shape)
end

defimpl Drawable, for: Point do
  def draw(%Point{x: x, y: y}) do
    IO.puts("Point at (#{x}, #{y})")
  end
end

defmodule Utils do
  @global_config %{debug: false}
  
  def format(str) do
    String.trim(str)
  end
  
  defmacro debug_print(msg) do
    quote do
      IO.puts("DEBUG: #{unquote(msg)}")
    end
  end
end

defmodule Main do
  alias Utils, as: U
  
  def main(args \\ []) do
    IO.puts("Hello, Elixir!")
  end
  
  defp add(a, b), do: a + b
end

File: tests/fixtures/simple.ts
// TypeScript test fixture
const VERSION = "1.0.0";

interface Point {
    x: number;
    y: number;
}

type Status = "ok" | "error";

enum Color {
    Red = 0,
    Green = 1,
    Blue = 2,
}

class Shape {
    constructor(public name: string) {}
    
    area(): number {
        return 0;
    }
}

function add(a: number, b: number): number {
    return a + b;
}

const multiply = (a: number, b: number): number => a * b;

export function main() {
    console.log("Hello, TypeScript!");
}

namespace Utils {
    export function format(str: string): string {
        return str.trim();
    }
}

type ComplexType<T> = {
    data: T;
    timestamp: Date;
};

let globalConfig: { debug: boolean } = {
    debug: false,
};

export default class DefaultExport {
    static readonly VERSION = "1.0.0";
}